### PR TITLE
Simplify isodate-traverse and explicitly recurse

### PR DESCRIPTION
**Pull request by jgershen**
This removes `component-each` as a dependency and removes a lot of code hedging around its behavior.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -11,65 +10,51 @@ var isodate = require('@segment/isodate');
...
-  if (type(input) === 'array') return array(input, strict);
+  if (strict === undefined)
+    strict = true;
+  if (type(input) === 'object') {
+    return traverseObject(input, strict);
+  } else if (type(input) === 'array') {
+    return traverseArray(input, strict);
+  } else if (isodate.is(input, strict)) {
+    return isodate.parse(input);
+  }
```
Comment (jlee9595) [inline lib/index.js]: I think a switch case on the value of type(input) where the default case contains an if (isodate.is(...)) would be cleaner here but it's just a matter of stylistic preference

**Comment by jlee9595** [comment]
simple is good, so much more readable now!

File: lib/index.js [hunk 2]
```diff
@@ -11,65 +10,51 @@ var isodate = require('@segment/isodate');
...
+ * @param {Boolean} strict - only convert strings with year, month, and date
  * @return {Object}
  */
 
 function traverse(input, strict) {
-  if (strict === undefined) strict = true;
-
-  if (type(input) === 'object') return object(input, strict);
-  if (type(input) === 'array') return array(input, strict);
+  if (strict === undefined)
```
Comment (jlee9595) [inline lib/index.js]: Looks like circle's complaining about no brackets here

File: lib/index.js (hunk 1 above)
Comment (jgershen) [reply lib/index.js]: I'm not a huge fan of `switch` personally so gonna skip :)

File: lib/index.js (hunk 2 above)
Comment (jgershen) [reply lib/index.js]: 👍

**Pull request merged**